trollmoves/tests/test_movers.py

```diff
@@ -88,6 +89,30 @@ def test_s3_copy_file_to_base(S3FileSystem):
     S3FileSystem.return_value.put.assert_called_once_with(ORIGIN, "data-bucket/filename.ext")
 
 
+@patch('trollmoves.movers.S3FileSystem')
+def test_s3_get_destination_from_string(S3FileSystem):
+    """Test get the correct S3-bucket destination filename."""
+    s3_mover = _get_s3_mover(ORIGIN, "s3://data-bucket/")
+    result = s3_mover._get_destination()
+    assert result == 'data-bucket/filename.ext'
+
+    s3_mover = _get_s3_mover(ORIGIN, "s3://data-bucket/upload/")
+    result = s3_mover._get_destination()
+    assert result == 'data-bucket/upload/filename.ext'
+
+    s3_mover = _get_s3_mover(ORIGIN, "s3://data-bucket/upload")
+    result = s3_mover._get_destination()
+    assert result == 'data-bucket/upload'
+
+
+@patch('trollmoves.movers.S3FileSystem')
+def test_s3_get_destination_from_parseresult(S3FileSystem):
+    """Test get the correct S3-bucket destination filename."""
+    s3_mover = _get_s3_mover(ORIGIN, urlparse("s3://data-bucket/upload/my_satellite_data.h5"))
+    result = s3_mover._get_destination()
+    assert result == 'data-bucket/upload//my_satellite_data.h5'
```

Comment:
There should not be double `//` in the expected result. In S3 `foo/bar` and `foo//bar` are different objects, and we clearly want the single `/` in there. Also see the comment above about checking the `.put()` arguments instead of the private method.

Reply:
That was a typo, thx!